integration-tests/src/main/java/org/apache/polaris/service/it/env/PolarisClient.java

```diff
@@ -0,0 +1,127 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.it.env;
+
+import static org.apache.polaris.service.it.ext.PolarisServerManagerLoader.polarisServerManager;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import jakarta.ws.rs.client.Client;
+import java.util.Random;
+import org.apache.iceberg.rest.RESTSerializers;
+import org.apache.polaris.core.admin.model.PrincipalWithCredentials;
+
+public final class PolarisClient implements AutoCloseable {
+  private final PolarisApiEndpoints endpoints;
+  private final Client client;
+  // Use an alphanumeric ID for widest compatibility in HTTP and SQL.
+  // Use MAX_RADIX for shorter output.
+  private final String clientId =
+      Long.toString(Math.abs(new Random().nextLong()), Character.MAX_RADIX);
+
+  private PolarisClient(PolarisApiEndpoints endpoints) {
+    this.endpoints = endpoints;
+
+    this.client = polarisServerManager().createClient();
+  }
+
+  public static PolarisClient polarisClient(PolarisApiEndpoints endpoints) {
+    return new PolarisClient(endpoints);
+  }
+
+  public static ObjectMapper buildObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    mapper.setPropertyNamingStrategy(new PropertyNamingStrategies.KebabCaseStrategy());
+    RESTSerializers.registerAll(mapper);
+    return mapper;
+  }
+
+  public String newEntityName(String hint) {
+    return polarisServerManager().transformEntityName(hint + "_" + clientId);
+  }
+
+  public ManagementApi managementApi(String authToken) {
+    return new ManagementApi(client, endpoints, authToken, endpoints.managementApiEndpoint());
+  }
+
+  public ManagementApi managementApi(ClientCredentials credentials) {
+    return managementApi(obtainToken(credentials));
+  }
+
+  public ManagementApi managementApi(PrincipalWithCredentials principal) {
+    return managementApi(obtainToken(principal));
+  }
+
+  public CatalogApi catalogApi(PrincipalWithCredentials principal) {
+    return new CatalogApi(
+        client, endpoints, obtainToken(principal), endpoints.catalogApiEndpoint());
+  }
+
+  public CatalogApi catalogApi(ClientCredentials credentials) {
+    return new CatalogApi(
+        client, endpoints, obtainToken(credentials), endpoints.catalogApiEndpoint());
+  }
+
+  public String obtainToken(PrincipalWithCredentials principal) {
+    return obtainToken(
+        new ClientCredentials(
+            principal.getCredentials().getClientId(),
+            principal.getCredentials().getClientSecret(),
+            "dummy-principal"));
```

Comment:
use the principal name?

Reply:
refactored to avoid dummy principal names